### PR TITLE
Add low-level-native-tls examples

### LINKED ISSUE #2853: Example for low-level-native-tls
- [x] I have looked for existing issues (including closed) about this

### Motivation

There are currently good examples for low-level-rustls and low-level-openssl that are sufficient to get a beginner started on their project, but their is no such example for native-tls. The ecosystem around acceptors, listeners and streams is confusing enough that there is a lot of potential value to a simple example providing certificate configuration and then serving via axum.

### Proposal

There should be a new example added providing the same functionality as low-level-rustls or low-level-openssl but with the native-tls backend.

### PR DESCRIPTION
**Pull request by ItsEthra**
Closes #2853

~~Shameless copy & paste of the rustls example.~~

### CODE REVIEW

**Comment by jing-951** [comment]
This is great

File: examples/low-level-native-tls/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                // We don't need to call `poll_ready` since `Router` is always ready.
+                tower_service.clone().call(request)
+            });
+
+            let ret = hyper_util::server::conn::auto::Builder::new(TokioExecutor::new())
+                .serve_connection_with_upgrades(stream, hyper_service)
+                .await;
+
+            if let Err(err) = ret {
+                warn!("error serving connection from {}: {}", addr, err);
```
Comment (jplatte) [inline examples/low-level-native-tls/src/main.rs:81]:
```suggestion
                warn!("error serving connection from {addr}: {err}");
```

File: examples/low-level-native-tls/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+publish = false
+
+[dependencies]
+axum = { path = "../../axum" }
+futures-util = { version = "0.3", default-features = false }
+hyper = { version = "1.0.0", features = ["full"] }
+hyper-util = { version = "0.1" }
+tokio = { version = "1", features = ["full"] }
+tokio-native-tls = "0.3.1"
+tower = { version = "0.4", features = ["make"] }
```
Comment (jplatte) [inline examples/low-level-native-tls/Cargo.toml:14]:
```suggestion
tower = { version = "0.5.1", features = ["make"] }
```

**Review by jplatte — APPROVED** [review]

**Comment by jplatte** [comment]
Thanks for the PR!

**Pull request merged**